integrations/criteo/lib/index.js

```diff
@@ -108,22 +107,29 @@ Criteo.prototype.track = function(track) {
 
 Criteo.prototype.productViewed = function(track) {
   var productId = track.productId() || '';
-  var event = [{ event: 'viewItem', product: productId }];
+  var event = [{ event: 'viewItem', item: productId }];
```

Comment:
This is still not correct. We are checking at the end of the function about `productId`, while it can be checked at the beginning and save time (look at  the example in the previous comment).

Reply:
Corrected, change the sequence though, for cases(**productId = '    '**):
```
if (typeof productId === 'string') {
  productId = productId.trim();
}

if (!productId) {
  // productId is madatory
  return;
}

var event = ...
```